From 6e533d5598ef875f30b84d931aae11b768465869 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Mon, 2 Jan 2023 15:00:02 +0100
Subject: [PATCH] meson.build: build introspection according to option only

The way the check is written, if the build is native, then the
introspection option has no effect.

Particularly yocto project does want to disable introspection in
native builds and enable it in cross builds (both via the option),
and without this patch the former is not possible.

Upstream-Status: Submitted [https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/5382]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index bfc33af0f6..94ffaa7769 100644
--- a/meson.build
+++ b/meson.build
@@ -854,7 +854,7 @@ endif
 
 # Introspection
 gir = find_program('g-ir-scanner', required : get_option('introspection'))
-build_gir = gir.found() and (not meson.is_cross_build() or get_option('introspection'))
+build_gir = gir.found() and get_option('introspection')
 
 subdir('gdk')
 subdir('gtk')
-- 
2.30.2

